### ISSUE TITLE
Random feedback on examples

### ISSUE THREAD

**Issue by icefoxen** [quinn-rs/quinn#84]
Random notes on things I find in the examples that are kind of hard to parse out, as one only moderately familiar with QUIC and tokio/futures:

 * A single-line description at the top of each would be nice, even if all it says is "see the readme for how to run the client and server"
 * The server example breaks out pieces into `handle_connection()` and `handle_request()`, which is nice, but the client example shoves everything into one big long gnarly future, which is hard to figure out.  When trying to understand how futures work, I often find breaking them up so you can actually see the type signatures to be really useful.
 * Sometimes the server uses `tokio_current_thread::spawn()` and sometimes `runtime.spawn()` which, since you use the single-threaded runtime, I *think* are equivalent?  It's not entirely clear though.
 * Also it *appears* that you must use the `current_thread` tokio `Runtime`, which sort of makes some sense I think but again isn't obvious to someone who doesn't know this.  Maybe it's documented somewhere, but I didn't find it.

These are all just the sorts  of things that are invisible to someone who knows what they're doing and really opaque to someone who doesn't.  I'll semi-happily update the examples to try to improve some of these things, if you guys want.

**Comment by Ralith** [comment]
Thanks for the feedback! That sort of PR is absolutely welcome; it's difficult to successfully take an outside perspective after being immersed in a code base for a long period, so deficiencies in this area are expected. While some of these problems arise due to the (present-day) design of tokio and futures, it's absolutely in-scope for our documentation to provide basic guidance on getting this stuff started.

> Also it appears that you must use the current_thread tokio Runtime, which sort of makes some sense I think but again isn't obvious to someone who doesn't know this. Maybe it's documented somewhere, but I didn't find it.

This is alluded to in the `quinn` crate rustdocs, but it's a bit oblique. @djc has at any rate talked me into implementing support for multithreaded runtimes, so this restriction is likely to go away soon after we finish updating to the latest IETF draft, unless it ends up being a major performance regression.

**Comment by icefoxen** [comment]
Also in the client example: `conn.close(0, b"done").map_err(|_| unreachable!())` <-- THAT line is the weirdest type-hack-thing I've ever seen and when trying to work off of that code it results in it being very unclear what is actually going on.

Mostly a note to myself so that I remember to try to untangle it later.  :-)

**Comment by Ralith** [comment]
I think everything discussed here has been addressed. Thanks for the feedback!
